### ISSUE TITLE
Test data for comparison with FreeType

### ISSUE THREAD

**Issue by dfrg** [googlefonts/fontations#203]
There are several test cases in `read-fonts` and `punchcut` where the condition is “produces results identical to FreeType.” The expected output is currently hand curated which is not ideal. We’d like to be able to generate this programmatically to both ensure consistent data and to facilitate adding new test cases. 

I see two potential solutions to this problem:

1. Take on the `freetype` crate as a test only dependency in these projects. Use this inline to generate expected values on the fly. 

2. Add a new crate to the repo that is used to generate the test data which is then included in the relevant places. 

I’m not sure if 1 violates our new dependency rule. I suppose we should clarify if build and test dependencies are subject to it.

**Comment by rsheeter** [comment]
A prospective "real user" is likely to need to compile our code so I would think that rules out build dependencies? 

Why not continue the current fontations model - one we've used in other places with considerable success - used for test fonts: submit both the output and the tooling, e.g. a script or a wee tool, to update the output. You would:

1. Submit a ttx and the compiled result of an "interesting" scenario, say a novel glyph setup
1. Run a script or wee tool to generate the freetype data for the same and emit it to a text file you commit
   * the prospective "real user" doesn't ever have to do this
1. In your test load the text file - the expected result - and the glyph punchcut gets from the font and compare them

**Comment by madig** [comment]
May I suggest comparative fuzzing, using FreeType as the oracle? 👀

**Comment by dfrg** [comment]
I suppose the “wee tool” would be equivalent to option 2? We ultimately need to execute FreeType code to generate the output so a script won’t suffice.

As for build and test dependencies, they’re simply additions to Cargo.toml that are only included during the relevant phases so shouldn’t add any friction for the user.

**Comment by rsheeter** [comment]
I agree it's basically the same as option 2, just removing the requirement to be a crate. For example, a wee python script or C++ tool become permissable.
